src/discussions/data/hooks.js

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable import/prefer-default-export */
+import { useContext, useEffect } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory, useLocation, useRouteMatch } from 'react-router';
+
+import { AppContext } from '@edx/frontend-platform/react';
+import { breakpoints, useWindowSize } from '@edx/paragon';
+
+import { Routes } from '../../data/constants';
+import { fetchCourseBlocks } from '../../data/thunks';
+import { clearRedirect } from '../posts/data';
+import { fetchCourseTopics } from '../topics/data/thunks';
+import { discussionsPath } from '../utils';
+import { selectAreThreadsFiltered } from './selectors';
+import { fetchCourseConfig } from './thunks';
+
+export function usePostThreadCount() {
+  return useSelector((state) => state.threads.totalThreads);
+}
+
+export function useTotalTopicThreadCount() {
+  const topics = useSelector((state) => state.topics.topics);
+
+  if (!topics) {
+    return 0;
+  }
+
+  return Object.keys(topics).reduce((total, topicId) => {
+    const topic = topics[topicId];
+    return total + topic.threadCounts.discussion + topic.threadCounts.question;
+  }, 0);
+}
+
+export const useSidebarVisible = () => {
+  const isFiltered = useSelector(selectAreThreadsFiltered);
+  const totalThreads = usePostThreadCount();
+  const isViewingTopics = useRouteMatch(Routes.TOPICS.PATH);
+
+  if (isFiltered) {
+    return true;
+  }
+
+  if (isViewingTopics) {
+    return true;
+  }
+
+  return totalThreads > 0;
+};
+
+export function useCourseTopicsAndBlocks(courseId) {
```

Comment:
This seems to be doing more than just topics and blocks, so maybe call it `useCourseDiscussionData`?

Reply:
Updated.